components/match2/commons/starcraft_starcraft2/opponent_starcraft.lua

```diff
@@ -0,0 +1,107 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Opponent/Starcraft
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Logic = require('Module:Logic')
+local Opponent = require('Module:Opponent')
+local StarcraftRace = require('Module:Race/Starcraft')
+local Table = require('Module:Table')
+local TypeUtil = require('Module:TypeUtil')
+
+local StarcraftOpponent = Table.deepCopy(Opponent)
+
+StarcraftOpponent.types.Player = TypeUtil.extendStruct(Opponent.types.Player, {
+	race = 'string?',
+})
+
+StarcraftOpponent.types.PartyOpponent = TypeUtil.struct{
+	isArchon = 'boolean',
+	--isSpecialArchon = 'boolean',
+	players = TypeUtil.array(StarcraftOpponent.types.Player),
+	type = TypeUtil.literalUnion(unpack(Opponent.partyTypes)),
+}
+
+StarcraftOpponent.types.Opponent = TypeUtil.union(
+	Opponent.types.TeamOpponent,
+	StarcraftOpponent.types.PartyOpponent,
+	Opponent.types.LiteralOpponent
+)
+
+--[[
+Not supported:
+
+Legacy TeamOpponent ({{TeamOpponent|players=...}})
+TeamOpponent without team template ({{TeamOpponent|name=...|short=...}})
+]]
+function StarcraftOpponent.readOpponentArgs(args)
+	local opponent = Opponent.readOpponentArgs(args)
+	local partySize = Opponent.partySize(opponent.type)
+
+	if partySize == 1 then
+		opponent.players[1].race = StarcraftRace.read(args.race)
```

Comment:
Should this be called `.faction` instead of `.race`? ie changing the lua key

Reply:
can be adressed in a sep. PR
this function is used by at least 3 (commons) + 6 (sc2) modules, so we have dependencies on the naming